packages/google-closure-compiler/lib/gulp/index.js

```diff
@@ -189,6 +189,12 @@ module.exports = function(initOptions) {
         // Convert these back to vinyl
         let outputFiles = [];
         if (stdOutData.trim().length > 0) {
+          if (code !== 0) {
+            this.emit('error', new PluginError(this.PLUGIN_NAME_, "Compiler error.\n" + stdOutData));
```

Comment:
Is this always standard out? Can it also be standard error? Should we report both?

Reply:
The error messages `Could not reserve enough space for object heap` and `Native memory allocation (malloc) failed` is sent to stdout. Error messages are not caught and falls into an exception on `outputFiles = JSON.parse(stdOutData);`

Reply:
I think you may need to combine both stdout and stderr then. Otherwise, if anything is written to stderr you are leaving it out as well.

Reply:
Thanks, added stderr to the output